stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+        raise ImportError("Jax is required")
+
+    if not can_make_gp:
+        raise ImportError("Tinygp is required to make the GP model.")
+
+    @jit
+    def likelihood_model(*args):
+        dict = {}
+        for i, params in enumerate(params_list):
+            if params[0:4] == "log_":
+                dict[params[4:]] = jnp.exp(args[i])
+            else:
+                dict[params] = args[i]
+        kernel = get_kernel(kernel_type=kernel_type, kernel_params=dict)
+        mean = get_mean(mean_type=mean_type, mean_params=dict)
+        gp = GaussianProcess(kernel, kwargs["Times"], mean_value=mean(kwargs["Times"]))
+        return gp.log_probability(kwargs["counts"])
+
+    return likelihood_model
+
+
+class GPResult:
+    """
+    Makes a GPResult object which takes in a Stingray.Lightcurve and samples parameters of a model
+    (Gaussian Process) based on the given prior and log_likelihood function.
+
+    Parameters
+    ----------
+    lc: Stingray.Lightcurve object
+        The lightcurve on which the bayesian inference is to be done
+
+    Other Parameters
+    ----------------
+    time : class: np.array
+        The array containing the times of the lightcurve
+
+    counts : class: np.array
+        The array containing the photon counts of the lightcurve
+
+    """
+
+    def __init__(self, Lc: Lightcurve) -> None:
+        self.lc = Lc
+        self.time = Lc.time
+        self.counts = Lc.counts
+        self.Result = None
+
+    def sample(self, prior_model=None, likelihood_model=None, max_samples=1e4):
+        """
+        Makes a Jaxns nested sampler over the Gaussian Process, given the
+        prior and likelihood model
+
+        Parameters
+        ----------
+        prior_model: jaxns.prior.PriorModelType object
+            A prior generator object
+
+        likelihood_model: jaxns.types.LikelihoodType object
+            A likelihood fucntion which takes in the arguments of the prior
+            model and returns the loglikelihood of the model
+
+        max_samples: int, default 1e4
+            The maximum number of samples to be taken by the nested sampler
+
+        Returns
+        ----------
+        Results: jaxns.results.NestedSamplerResults object
+            The results of the nested sampling process
+
+        """
+        if not jax_avail:
+            raise ImportError("Jax is required")
+
+        if not can_sample:
+            raise ImportError("Jaxns not installed! Can't sample!")
+
+        self.prior_model = prior_model
+        self.likelihood_model = likelihood_model
+
+        NSmodel = Model(prior_model=self.prior_model, log_likelihood=self.likelihood_model)
+        NSmodel.sanity_check(random.PRNGKey(10), S=100)
+
+        self.Exact_ns = ExactNestedSampler(NSmodel, num_live_points=500, max_samples=max_samples)
+        Termination_reason, State = self.Exact_ns(
+            random.PRNGKey(42), term_cond=TerminationCondition(live_evidence_frac=1e-4)
+        )
```

Comment:
I think it would be useful to print the `Termination_reason` (also lowercase) here, if it's a useful message, because it would help the user identify whether things went wrong or not.